src/asciidoctor-extensions-api.js

```diff
@@ -292,3 +295,119 @@ var DocinfoProcessor = Extensions.DocinfoProcessor;
 DocinfoProcessor.$$proto.atLocation = function (value) {
   this.$at_location(value);
 };
+
+function initializeProcessorClass (scope, functions) {
+  var postConstructFunction;
+  var initializeFunction;
+  var isHandlesFunctionDefined = false;
+  for (var key in functions) {
+    if (functions.hasOwnProperty(key)) {
+      (function (key) {
+        var userFunction = functions[key];
+        if (key === 'postConstruct') {
+          postConstructFunction = userFunction;
+        } else if (key === 'initialize') {
+          initializeFunction = userFunction;
+        } else {
+          if (key === 'handles?') {
+            isHandlesFunctionDefined = true;
+          }
+          Opal.defn(scope, '$' + key, function () {
+            userFunction.apply(this, arguments);
+          });
+        }
+      }(key));
+    }
+  }
+  var initialize;
+  if (typeof initializeFunction === 'function') {
+    initialize = function () {
+      initializeFunction.apply(this, arguments);
+      if (typeof postConstructFunction === 'function') {
+        postConstructFunction.bind(this)();
+      }
+    };
+  } else {
+    initialize = function () {
+      Opal.send(this, Opal.find_super_dispatcher(this, 'initialize', initialize));
+      if (typeof postConstructFunction === 'function') {
+        postConstructFunction.bind(this)();
+      }
+    };
+  }
+  Opal.defn(scope, '$initialize', initialize);
+  Opal.defn(scope, 'super', function (func) {
+    if (typeof func === 'function') {
+      Opal.send(this, Opal.find_super_dispatcher(this, func.name, func));
+    } else {
+      // Bind the initialize function to super();
+      Opal.send(this, Opal.find_super_dispatcher(this, 'initialize', initialize));
+    }
+  });
+  if (!isHandlesFunctionDefined) {
+    Opal.defn(scope, '$handles?', function () {
+      return true;
+    });
+  }
+}
+
+Extensions.createIncludeProcessor = function (name, functions) {
+  var superclass = Opal.const_get_qualified(Extensions, 'IncludeProcessor');
+  var scope = Opal.klass(Opal.Object, superclass, name, function () {});
+  initializeProcessorClass(scope, functions);
+  return scope;
+};
+
+Extensions.newIncludeProcessor = function (name, functions) {
```

Comment:
@mojavelinux What do you think about these functions ? The only purpose is to hide `$new()`... do you have a better idea ? should we remove these functions for now ?